views/default/input/date.php

```diff
@@ -48,8 +54,21 @@
 	$vars['value'] = gmdate('Y-m-d', $vars['value']);
 }
 
+if (isset($vars['datepicker_options'])) {
+	$vars['data-datepicker-opts'] = json_encode($vars['datepicker_options']);
+	unset($vars['datepicker_options']);
+}
+
 echo elgg_format_element('input', $vars);
 
-if (elgg_is_xhr()) {
-	echo elgg_format_element('script', null, 'elgg.ui.initDatePicker();');
+if (isset($vars['id'])) {
+	$selector = "#{$vars['id']}";
+} else {
+	$selector = ".elgg-input-date[name='{$vars['name']}']";
 }
+?>
+<script>
+	require(['input/date'], function(datepicker) {
+		datepicker.init('<?= $selector ?>');
```

Comment:
Note this might result in a broken string:
```js
datepicker.init('.elgg-input-date[name='myname']');
```
Solution is to always use json_encode() to write JS string literals.